R/PMP.R

```diff
@@ -0,0 +1,315 @@
+#'Estimate the Probable Maximum Precipitation based on Herschfield Methodology
+#'
+#'PMP Estimation based on Hershfield methodology from Daily values
+#'@param val.max        Vector with annual maximum.
+#'@param input.hr      PMP to be defined 24 hrs, 6 hrs or 1 hr.
+#'@param area.km     Watershed area. This methodology is for Western US. By default, if the correction is not needed is recommended a value of 1.
+#'@param number.of.obs.units How many observation units are implemented. ie Daily info for 24 hrs PMP is 1.
+#'@param graph    True/False if the graphical output is needed.
+#'
+#'@return
+#' List Elements:
+#' \itemize{
+#' \item values - maximum values used in the analysis
+#' \item n - ammount of years
+#' \item correction - data.frame with the parameters and corrections for average and St. Dev.
+#' \item Km - corrections for rainfal duration and mean annual series (based on Figure 4.1)
+#' \item PMP - probably maximum precipitation estimated. This value does not include correction from Figure 4.7, which is aplicable to Western US.
+#' \item PMP.with.area.correction.for.West.US - PMP value corrected based on watershed area based on Figure 4.7
+#' }
+#' Please note that data from NA are removed in the analysis.
+#'@examples
+#' ## EX 1: PMP for the daily maximum values for a 24 hours PMP in an area of 200km2 (Optional for Western US).
+#' ## Random 30 number from 60 to 150
+#' PMP(val.max = runif(30,min = 60,150),input.hrs = 24,number.of.obs.unit = 1,area.km = 200,graphs = T)
+#' @references [WMO] World Meteorological Organization (2009), Manual on Estimation of Probable Maximum Precipitation (PMP) WMO-No 1045
+#'
+#'@author Victor Munoz
+#'@export
+PMP<-function(val.max,input.hrs=24,number.of.obs.unit=1,area.km=1,graphs=T){
+
+        if(graphs==T){
+
+                require(ggplot2)
+                require(gridExtra)
+
+        }
+
+        if(is.na(match(input.hrs,c(24,6,1)))){
+                return(print("input.hrs should be 24, 6 or 1 hr."))
+        }
+
+
+        if(number.of.obs.unit<=0){
+                return(print("number.of.obs.unit should be positive"))
+        }
+
+
+        if(!round(number.of.obs.unit)-number.of.obs.unit==0){
+                return(print("number.of.obs.unit should be integer"))
+        }
+
+
+        Max.year<-val.max
+
+        #remove NA
+        X.n<-Max.year[!is.na(Max.year)]
+
+        X.max.location<-match(max(X.n),X.n)
+
+        X.n_m<-X.n[-X.max.location]
+
+        n_PMP<-length(X.n)
+        n_PMP<<-n_PMP
+        X<-list(c(X.n),c(X.n_m))
+
+        #Estimations of mean and sd
+        X.mean<-lapply(X,mean)
+        X.sd<-lapply(X,sd)
+
+        Xn_m_div<-X.mean[[2]]/X.mean[[1]]
+        Sn_m_div<-X.sd[[2]]/X.sd[[1]]
+
+        #Fig 4.2
+        Fig_4.2f<-function(n_PMP,Xn_m_div){

```

Comment:
Here you have defined a function within another function. It is preferable to define that function outside the main function and then call it when needed in other functions. Also see the general comment about function naming.

Reply:
the function is just specific for the transformation from the graph to the equation. So I really don't consider appropriate to include the function outside, I can be renamed; but it won't be used in any other place. I will really suggest just leave it, but renaming it.

Reply:
@vmunozs I still think that it is easier in the long run to not define functions within other functions. There is really no downside to cutting them out from this above function, pasting them below in the same .R file and then calling them in the larger function. This makes you code more readable and makes debugging a bit easier.